src/HealthChecks.Kafka/DependencyInjection/KafkaHealthCheckBuilderExtensions.cs

```diff
@@ -113,4 +113,36 @@ public static IHealthChecksBuilder AddKafka(
             tags,
             timeout));
     }
+
+    /// <summary>
+    /// Add a health check for Kafka cluster.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
+    /// <param name="optionFactory">Options to configure Kafka health check.</param>
+    /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'kafka' will be used for the name.</param>
+    /// <param name="failureStatus">
+    /// The <see cref="HealthStatus"/> that should be reported when the health check fails. Optional. If <c>null</c> then
+    /// the default status of <see cref="HealthStatus.Unhealthy"/> will be reported.
+    /// </param>
+    /// <param name="tags">A list of tags that can be used to filter sets of health checks. Optional.</param>
+    /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
+    /// <returns>The specified <paramref name="builder"/>.</returns>
+    public static IHealthChecksBuilder AddKafka(

```

Comment:
@mitchdenny would adding this overload solve the problem you have faced in Aspire?

I can see that you have solved it by simply reusing the ctor of the health check and registering your own thing in the DI container:

https://github.com/dotnet/aspire/blob/422d777c8dc7df154d9eea268f0d28a35e673f1b/src/Components/Aspire.Confluent.Kafka/AspireKafkaProducerExtensions.cs#L139-L151